src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * This function simulates a chat bot who interacts with a user to keep track of a todo list.
+ *
+ * @author Megan Wee Rui En
+ * @version CS2103T AY21/22 Semester 1
+ */
+package duke;
+
+import duke.command.Command;
+import duke.exceptions.CommandDoesNotExist;
+import duke.exceptions.DukeExceptions;
+import duke.parser.Parser;
+import duke.storage.Storage;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Duke {
+
+    private final Storage storage;
+    private final TaskList taskList;
+    private final Ui ui;
+    private final Command command;
+
+    public Duke(String filePath) throws IOException {
+        this.ui = new Ui(); // Performs the self introduction upon successful initialization.
+        this.storage = new Storage(filePath);
+        this.taskList = new TaskList(storage.load());
+        this.command = new Command(storage, ui);
+    }
+
+    /**
+     * This function is the body of the chat bot, and contains the flow and commands.
+     */
+    public void run() throws DukeExceptions, IOException {
+        ui.showWelcome();
+        String user_command; // this is the container for the full command received from the user
+        String cmd; // this is the container for the first word of the command received from the user
+
+        ///// This listens for the commands and interprets them
+        // This part listens for user input and repeats until the command "bye" is identified
```

Comment:
```suggestion
// Listens for user input and repeats until the command "bye" is identified
```
I think briefer comments would be an improvement to comment quality.